git: Add Depth to SubmoduleUpdateOptions

matejrisek opened the pull request:
In order to allow shallow cloning for submodules we added the new depth option to the SubmoduleUpdateOptions. This option is then being used while fetching a submodule.

pjbgf commented:
@matejrisek please squash the commits and re-word the commit message based on the [Contributing Guidelines](https://github.com/go-git/go-git/blob/master/CONTRIBUTING.md) (e.g. new PR title).

matejrisek commented:
> @matejrisek please squash the commits and re-word the commit message based on the [Contributing Guidelines](https://github.com/go-git/go-git/blob/master/CONTRIBUTING.md) (e.g. new PR title).

Hey @pjbgf thanks for the pointers - I've squashed the commits and changed the commit message. Could you please verify it's ok now.

pjbgf reviewed (approved):
@matejrisek thanks for working on this! :bow:
